src/vos/vos_pool.c

```diff
@@ -1272,6 +1272,26 @@ vos_pool_create_ex(const char *path, uuid_t uuid, daos_size_t scm_sz, daos_size_
 	return rc;
 }
 
+int
+vos_pool_roundup_size(daos_size_t *scm_sz, daos_size_t *meta_sz)
+{
+	int    backend;
+	size_t alignsz;
+
+	backend = umempobj_get_backend_type();
+	if ((*scm_sz != *meta_sz) && (backend == DAOS_MD_BMEM))
+		backend = DAOS_MD_BMEM_V2;
+
+	/* Round up the size such that it is compatible with backend */
+	alignsz = umempobj_pgsz(backend);
+
+	*scm_sz = D_ALIGNUP(*scm_sz, alignsz);
+	if (*meta_sz)
```

Comment:
Looks like control plane may pass zero meta_sz? Then we could mistakenly assume it's BMEM_V2 backend in above code?

BTW, I think we'd move the "max(tca->tca_scm_size / dss_tgt_nr, 1 << 24)" from tgt_create_preallocate() to here and apply it to both scm size and meta size, right?

Reply:
meta_sz check was added assuming that for pmem mode this may not be set. Verified that that both scm_sz and meta_sz is set by the control plane in all modes pmem, bmem_v1 and bmem_v2. Hence removed the checks and added asserts.